service/resource/configmap/configmap.go

```diff
@@ -52,6 +57,9 @@ func New(config Config) (*Resource, error) {
 	if config.Logger == nil {
 		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
 	}
+	if config.Prometheus == nil {
```

Comment:
I would maybe call this prometheus reloader. It is not really clear what `Prometheus` really means in this case. Up to you.

Reply:
I agree it feels a little bit like home grows prom client.

Reply:
I like `PrometheusReloader` more, updated.